pkg/controlplane/instance_test.go

```diff
@@ -350,6 +350,20 @@ func TestNoBetaVersionsEnabledByDefault(t *testing.T) {
 			t.Errorf("Beta API version %s enabled by default", gv.String())
 		}
 	}
+
+	for gvr, enabled := range config.ResourceConfigs {
+		if !strings.Contains(gvr.Version, "beta") || !enabled {
+			continue
+		}
+
+		// we have enabled a beta api by resource {g,v,r}, we also expect the
+		// beta api by version {g,v} to be disabled. This is so a programmer
+		// remembers to add the new beta version to betaAPIGroupVersionsDisabledByDefault.
```

Comment:
do we need the same check in TestNoAlphaVersionsEnabledByDefault to make sure those are registered in alphaAPIGroupVersionsDisabledByDefault?

Reply:
I added it for alpha as well, and to ensure the default `vars` always have the proper version, I added `TestDefaultVars`.